app/scripts/lib/metrics.js

```diff
@@ -153,6 +159,66 @@ define(function (require, exports, module) {
       this._clearInactivityFlushTimeout();
     },
 
+    notifications: {
+      /* eslint-disable sorting/sort-object-props */
+      'flow.initialize': '_initializeFlowModel',
+      'flow.event': '_logFlowEvent'
+      /* eslint-enable sorting/sort-object-props */
+    },
+
+    /**
+     * Initialize the flow model. If it's already been initalized, do nothing.
+     * Initialization may fail if the required flow properties can't be found,
+     * either in the DOM or the resume token.
+     */
+    _initializeFlowModel () {
+      if (this._flowModel) {
+        return;
+      }
+
+      const flowModel = new Flow({
+        sentryMetrics: this._sentryMetrics,
+        window: this._window
+      });
+
+      if (flowModel.has('flowId')) {
+        this._flowModel = flowModel;
+      }
+    },
+
+    /**
+     * Log a flow event. If there is no flow model, do nothing.
+     *
+     * @param {Object} data
+     * Event data.
+     *
+     *   @param {String} data.event
```

Comment:
Okay, I'll revert this bit. Fwiw, I dislike our preferred style because the param name collides with the description text when you're viewing the code. I find it harder to read.